src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java

```diff
@@ -51,8 +51,11 @@
 /**
  * A Java Bean that implements the component lifecycle interfaces by
  * calling the bean's setters for all configuration attributes.
+ *
+ * @deprecated since 10.3.4. Use {@code AbstractAutomaticBean} instead.
  */
 // -@cs[AbstractClassName] We can not break compatibility with previous versions.
+@Deprecated(since = "10.3.4")
 public abstract class AutomaticBean
```

Comment:
https://github.com/checkstyle/checkstyle/issues/5337#issuecomment-464571041
https://github.com/checkstyle/checkstyle/issues/5337#issuecomment-1094370256

Why can't we rename `AutomaticBean` to `AbstractAutomaticBean` and create a new `AutomaticBean` that extends `AbstractAutomaticBean` but is marked as deprecated?

With this type of make up, we won't need additional "ifs" and current code can just swap to `AbstractAutomaticBean` as both extend it.

Reply:
I see no problem. Never would have came up with this though.

Reply:
@stoyanK7 Let us know when this is done.

Reply:
@rnveach it is. I'm trying to address the failing CI jobs and I will tag you then

Reply:
Done.